### PR TITLE
feat: Add slot for additional buttons

### PR DESCRIPTION
**Pull request by PeterDraex**
## What kind of change does this pull request introduce?

This PR enables users to add more buttons to the preview area. They will be rendered right before buttons for "Refresh" and "Open in CodeSandbox".

## What is the current behavior?

Users can't add custom buttons to preview.

## What is the new behavior?

User can add a button in the following way:
```jsx
<SandpackPreview
    additionalButtons={
      <button
        className="sp-button"
        style={{ padding: 'var(--sp-space-1) var(--sp-space-3)' }}
        onClick={() => window.alert('Bug reported!')}>
        Report bug
      </button>
    }
  />
```

## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

Manually tested in the documentation and in a personal project.

## Checklist

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged

### CODE REVIEW

File: sandpack-react/src/components/Preview/index.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { useClasser } from "@code-hike/classer";
 import * as React from "react";
 
+import type { SandpackMessage } from "../../../../sandpack-client";
```
Comment (danilowoz) [inline sandpack-react/src/components/Preview/index.tsx:4]:
```suggestion
import type { SandpackMessage } from "@codesandbox/sandpack-client";
```

File: sandpack-react/src/components/Preview/index.tsx [hunk 2]
```diff
@@ -32,6 +33,7 @@ export interface PreviewProps {
   showOpenInCodeSandbox?: boolean;
   showRefreshButton?: boolean;
   showSandpackErrorOverlay?: boolean;
+  additionalButtons?: JSX.Element;
```
Comment (danilowoz) [inline sandpack-react/src/components/Preview/index.tsx:36]: I'm not fully confident with the prop name. What about?
- Use `children` instead?
- `renderActionSection`;
- `renderActions`;
- `renderChildrenActions`;
- `actionsChildren`

What do you think?

File: website/docs/docs/advanced-usage/components.md [hunk 3]
```diff
@@ -111,6 +111,43 @@ There's nothing stopping you from rendering multiple previews in the same `Provi
   </SandpackLayout>
 </SandpackProvider>
 
+## Additional buttons
```
Comment (danilowoz) [inline website/docs/docs/advanced-usage/components.md:114]: I think it should be a subitem from Preview
```suggestion
### Additional buttons
```

File: sandpack-react/src/components/Preview/index.tsx (hunk 1 above)
Comment (PeterDraex) [reply sandpack-react/src/components/Preview/index.tsx:4]: I've tried that at first, but VS Code shows the following error message:
> Could not find a declaration file for module '@codesandbox/sandpack-client'. '/home/peter/sandpack/sandpack-client/dist/cjs/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/codesandbox__sandpack-client` if it exists or add a new declaration (.d.ts) file containing `declare module '@codesandbox/sandpack-client';`ts(7016)

Comment (danilowoz) [reply sandpack-react/src/components/Preview/index.tsx:4]: I think you should run `yarn build` at first, in order to generate the types

Comment (PeterDraex) [reply sandpack-react/src/components/Preview/index.tsx]: Thanks :)

File: website/docs/docs/advanced-usage/components.md [hunk 4]
```diff
@@ -111,6 +111,43 @@ There's nothing stopping you from rendering multiple previews in the same `Provi
...
+          Report bug
+        </button>
+      }
+    />
+    <SandpackCodeEditor />
+  </SandpackLayout>
+</SandpackProvider>
+```
+
+<SandpackProvider template="react">
```
Comment (danilowoz) [inline website/docs/docs/advanced-usage/components.md:135]: Just noticed, is this duplicated?

**Comment by danilowoz** [comment]
🚀 Thanks a lot @PeterDraex, awesome work here!

**Pull request merged**